staging/src/k8s.io/client-go/discovery/restmapper.go

```diff
@@ -144,7 +144,10 @@ func NewRESTMapper(groupResources []*APIGroupResources, versionInterfaces meta.V
 func GetAPIGroupResources(cl DiscoveryInterface) ([]*APIGroupResources, error) {
 	apiGroups, err := cl.ServerGroups()
 	if err != nil {
-		return nil, err
+		if apiGroups == nil || len(apiGroups.Groups) == 0 {
```

Comment:
When `err` is not nil,  we should let `apiGroups` always be nil. Right?

And the code here just shows `apiGroup==nil` when `err != nil`.
 https://github.com/kubernetes/kubernetes/blob/bb3887d7ddc789a8d126e3c5c84a3eaacddd08bf/staging/src/k8s.io/client-go/discovery/discovery_client.go#L129-L157

Reply:
that is only one possible implementation of DiscoveryInterface. GetAPIGroupResources should not make assumptions about the implementation, and already assumes results can be partial.